### PR TITLE
[#86]: Adds `aws-file` as connection method

### LINKED ISSUE #86: [CT-1936] Support Google Cloud Auth from AWS Credentials File
### Describe the feature
Support connecting to BigQuery via from an AWS Credentials File.

### Describe alternatives you've considered
None

### Additional context
[Google Cloud Auth  AWS Docs](https://googleapis.dev/python/google-auth/latest/reference/google.auth.aws.html)

This is how my org does gcloud auth, so we don’t have many alternatives here

### Who will this benefit?
Any user/company who runs Google cloud workloads from AWS and keeps credentials in AWS.

### Are you interested in contributing this feature?
Yes

### PR DESCRIPTION
**Pull request by siephen**
resolves #86 

### Description
Adds `aws-file` as a connection method option for getting credentials from AWS. 
### Checklist

- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [ ] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] I have updated the `CHANGELOG.md` and added information about my change to the "dbt-bigquery next" section.

### CODE REVIEW

**Comment by McKnight-42** [comment]
@siephen if you could sign the cla that should kick off some tests, we will have a lot of failures for now till we can incorporate This connection type into CI/CD

**Comment by siephen** [comment]
> @siephen if you could sign the cla that should kick off some tests, we will have a lot of failures for now till we can incorporate This connection type into CI/CD

I've signed the CLA and getting details on how to get this working locally to ensure we can replicate it within the CI/CD pipeline.

**Comment by siephen** [comment]
> Thank you for your pull request and welcome to our community. We could not parse the GitHub identity of the following contributors: **Kyle Wigley**. This is most likely caused by a git client misconfiguration; please make sure to:
> 
> 1. check if your git client is configured with an email to sign commits `git config --list | grep email`
> 2. If not, set it up using `git config --global user.email email@example.com`
> 3. Make sure that the git commit email is configured in your GitHub account settings, see https://github.com/settings/emails

@McKnight-42 , any ideas why this is hitting after I merged the main branch into it?

**Comment by siephen** [comment]
@McKnight-42/@VersusFacit, if I add a screenshot of a dbt debug with this and provide the details vis gist, is that enough to get this under review?

**Comment by siephen** [comment]
@nathaniel-may , looks like the CLI bot removed the tag after I updated the branch with master, Can you add it back in, and this PR is ready for some feedback.

**Comment by siephen** [comment]
> Thank you for your pull request and welcome to our community. We could not parse the GitHub identity of the following contributors: **Kyle Wigley**. This is most likely caused by a git client misconfiguration; please make sure to:
> 
> 1. check if your git client is configured with an email to sign commits `git config --list | grep email`
> 2. If not, set it up using `git config --global user.email email@example.com`
> 3. Make sure that the git commit email is configured in your GitHub account settings, see https://github.com/settings/emails

@McKnight-42 or @nathaniel-may , I do not understand why this comment is coming up, and also it seems the branch keeps going stale and then `cla-bot` removes the `cla:yes` label.

**Comment by Fleid** [comment]
Hey @McKnight-42 would you say this one is `ready-for-review`?

**Comment by siephen** [comment]
@McKnight-42 or @Fleid, I'm having issues running this locally with this set up, so I'm unsure what the issue is, I'm going to keep working on this this week, but I was curious if y'all have seen anything related to [service account impersonation](https://docs.getdbt.com/reference/warehouse-setups/bigquery-setup#service-account-impersonation) that could be used with the aws auth process in Google Cloud

File: dbt/adapters/bigquery/connections.py [hunk 1]
```diff
@@ -304,6 +304,16 @@ def get_google_credentials(cls, profile_credentials) -> GoogleCredentials:
...
 
+        elif method == BigQueryConnectionMethod.AWS_FILE:
+            aws_creds = GoogleAuthAWS.Credentials
+            details = profile_credentials.keyfile
+            return aws_creds.from_file(details, scopes=profile_credentials.scopes)
+
+        elif method == BigQueryConnectionMethod.AWS_FILE:
+            aws_creds = GoogleAuthAWS.Credentials
+            details = profile_credentials.keyfile
+            return aws_creds.from_file(details, scopes=profile_credentials.scopes)
```
Comment (Fleid) [inline dbt/adapters/bigquery/connections.py:315]: Am I missing something or is this `elif` block repeated as is?

**Comment by Fleid** [comment]
Hi @siephen,

Are you have trouble resolving the CLA warning? Have you followed the step the bot is suggesting? We won't be able to merge the PR as-is if not.

To your question, let me start by restating that I'm not an expert at Google authentication ;)

To me that [impersonation option](https://docs.getdbt.com/reference/warehouse-setups/bigquery-setup#service-account-impersonation) leverages the local OAuth authentication context to be resolved. 
- Unattended, the identity your runner (CI agent) is using needs to be able to impersonate the service account you define there
- Locally you should be able to use the [gcloud option ](https://docs.getdbt.com/reference/warehouse-setups/bigquery-setup#oauth-via-gcloud)to set you up properly

That's my current understanding anyway - please correct me if I'm wrong ;)

**Comment by siephen** [comment]
@Fleid, Every time I update the branch with the new contents from main. CLA warning shows for `Kyle Wigley`: I'm not sure how to resolve that, without destroying this branch and starting over.

Do you have any ideas on how to resolve that?

File: dbt/adapters/bigquery/connections.py (hunk 1 above)
Comment (siephen) [reply dbt/adapters/bigquery/connections.py:315]: It does look duplicated. I'll resolve that.

Comment (siephen) [reply dbt/adapters/bigquery/connections.py]: removed in the latest push.

**Comment by Fleid** [comment]
What shows if you `git config --list | grep email` in your local terminal?

**Comment by Fleid** [comment]
If that was me, I would start over with a new PR, based of the latest main.
I'm sure there is a combination of git magic to get us out of the current situation, but I'm nowhere near that power level 🧙

**Comment by VersusFacit** [comment]
This sort of thing has happened in the past and has been remedied with a new PR as annoying as that can be. Just link the new PR to this one so there's some semblance of history. That way they're at least connected and not lost in the ether from one another.

**Comment by Fleid** [comment]
@siephen, I'm going to close this PR.
Please open a new one from a fresh branch off main, and we'll move other there.
Sorry about the clunky process :/

**Pull request closed without merging**